projects/demo/src/app/form1/form1.component.html

```diff
@@ -0,0 +1,55 @@
+<ng-container  *ngIf="data$ |async as data">
+<form  validationId="sample-data" #form="ngForm">
+  <input
+    type="hidden"
+    name="id"
+    [(ngModel)]="data.id"
+  >
+  <label>
+    <span>Name</span>
+    <input
+      type="text"
+      name="name"
+      placeholder="Your name"
+      [(ngModel)]="data.name"
+    />
+  </label>
+  <label>
+    <span>Email</span>
+    <input
+      type="email"
+      name="email"
+      placeholder="Your email"
+      [(ngModel)]="data.email"
+    />
+  </label>
+  <label>
+    <span>Date of Birth</span>
+    <input
+      type="date"
+      name="dob"
+      placeholder="Your date of birth"
+      [(ngModel)]="data.dob"
+    >
+  </label>
+  <label>
+    <span>Password</span>
+    <input
+      type="password"
+      name="password"
+      placeholder="Your password"
+      [(ngModel)]="data.password"
+    >
+  </label>
+  <label>
+    <span>Confirm password</span>
+    <input
+      type="password"
+      name="confirm"
+      placeholder="Your password again"
+      [(ngModel)]="data.confirm"
+    >
+  </label>
+  <input type="submit" value="Submit" [disabled]="form.invalid">
+  <button type="reset">Reset</button>
```

Comment:
Was this on purpose left without any action?

Reply:
yes, I'm not yet handling the actual submit.

Reply:
added one anyway